In [1]:
import pandas as pd
import numpy as np
import geopandas as gp
import shapely
from planet4 import io, markings

In [2]:
from coverage_utils import CoverageCalculator, get_obsids_for_region
from my_io import get_tilecoords, get_metadata, get_region_names

In [3]:
olddata = pd.read_csv("n_obj_df.csv")

In [4]:
olddata

Latitude       Ls  MY     Which  fraction  n_objects            obsid  \
0    -85.4074  214.785  29      both     0.115        933  ESP_012079_0945   
1    -85.4065  183.727  29      both     0.070       1971  ESP_011407_0945   
2    -85.4056  198.040  29      both     0.072       1525  ESP_011723_0945   
3    -85.4049  207.892  29      both     0.168        788  ESP_011934_0945   
4    -85.4067  235.282  29      both     0.111       1285  ESP_012501_0945   
5    -85.4117  224.964  29      both     0.087        350  ESP_012290_0945   
6    -85.4086  183.683  29      both     0.074        453  ESP_011406_0945   
7    -85.4074  214.785  29      fans     0.088        781  ESP_012079_0945   
8    -85.4065  183.727  29      fans     0.046       1568  ESP_011407_0945   
9    -85.4056  198.040  29      fans     0.047        939  ESP_011723_0945   
10   -85.4049  207.892  29      fans     0.126        678  ESP_011934_0945   
11   -85.4067  235.282  29      fans     0.097       1253  ESP_012501_0945   
12   -85.4117  224.964  29      fans     0.083        326  ESP_012290_0945   
13   -85.4086  183.683  29      fans     0.026        291  ESP_011406_0945   
14   -85.4074  214.785  29  blotches     0.037        331  ESP_012079_0945   
15   -85.4065  183.727  29  blotches     0.029        529  ESP_011407_0945   
16   -85.4056  198.040  29  blotches     0.032        808  ESP_011723_0945   
17   -85.4049  207.892  29  blotches     0.057        230  ESP_011934_0945   
18   -85.4067  235.282  29  blotches     0.021        131  ESP_012501_0945   
19   -85.4117  224.964  29  blotches     0.006         47  ESP_012290_0945   
20   -85.4086  183.683  29  blotches     0.056        256  ESP_011406_0945   
21   -85.4041  243.978  30      both     0.049       2256  ESP_021494_0945   
22   -85.4071  209.383  30      both     0.075        209  ESP_020782_0945   
23   -85.4086  206.269  30      both     0.062        220  ESP_020716_0945   
24   -85.4078  199.501  30      both     0.065        560  ESP_020571_0945   
25   -85.4105  186.879  30      both     0.080        445  ESP_020294_0945   
26   -85.4075  216.293  30      both     0.079        164  ESP_020927_0945   
27   -85.4056  184.557  30      both     0.054        481  ESP_020242_0945   
28   -85.4041  243.978  30      fans     0.032       1053  ESP_021494_0945   
29   -85.4071  209.383  30      fans     0.050        145  ESP_020782_0945   
..        ...      ...  ..       ...       ...        ...              ...   
624  -86.2597  214.426  30      fans     0.033         53  ESP_020888_0935   
625  -86.2599  197.699  30  blotches     0.025         97  ESP_020532_0935   
626  -86.2535  242.693  30  blotches     0.027       2392  ESP_021468_0935   
627  -86.2587  185.136  30  blotches     0.058        208  ESP_020255_0935   
628  -86.2751  285.423  30  blotches     0.001         27  ESP_022339_0935   
629  -86.2597  214.426  30  blotches     0.011         36  ESP_020888_0935   
630  -85.0057  207.279  29      both     0.056        216  ESP_011921_0950   
631  -85.0057  207.279  29      fans     0.055        201  ESP_011921_0950   
632  -85.0057  207.279  29  blotches     0.002         30  ESP_011921_0950   
633  -73.4875  223.218  29      both     0.014       2024  ESP_012254_1065   
634  -73.4787  254.344  29      both     0.031       2805  ESP_012887_1065   
635  -73.4875  223.218  29      fans     0.004        399  ESP_012254_1065   
636  -73.4787  254.344  29      fans     0.002         31  ESP_012887_1065   
637  -73.4875  223.218  29  blotches     0.011       1740  ESP_012254_1065   
638  -73.4787  254.344  29  blotches     0.030       2790  ESP_012887_1065   
639  -86.9844  212.826  29      both     0.019        141  ESP_012038_0930   
640  -86.9850  243.661  29      both     0.010        363  ESP_012671_0930   
641  -86.9846  250.830  29      both     0.007        400  ESP_012816_0930   
642  -86.9844  212.826  29      fans     0.018        100  ESP_012038_0930   
643  -86.9850  243.661  29 

In [5]:
data = pd.read_csv("P4_catalog_v1.1_tile_coords_final.csv")

In [286]:
7865/3

2621.6666666666665

In [285]:
data.BodyFixedCoordinateX.value_counts(dropna=False)

NaN            7865
 219.705037       1
 77.142883        1
 221.775759       1
-65.686409        1
-54.445869        1
 222.498339       1
 476.633376       1
 468.072076       1
 477.416816       1
-155.231403       1
 218.922983       1
-106.578993       1
 10.630504        1
-327.961453       1
 14.381824        1
 220.281940       1
-155.079249       1
 222.612662       1
-372.455625       1
 467.510836       1
 896.267656       1
 221.387972       1
-109.896246       1
 238.464911       1
-54.085377        1
 221.025511       1
 196.595188       1
-181.098864       1
 109.834441       1
               ... 
 471.454036       1
-895.647612       1
-371.350736       1
 901.091782       1
 197.277169       1
-50.640603        1
 9.544266         1
 221.827307       1
-55.829695        1
 221.183422       1
 112.117200       1
-155.429163       1
-284.213363       1
 113.308110       1
-188.246681       1
 220.529718       1
-26.676126        1
 470.120394       1
-66.944225        1


In [7]:
#x1 = data[data['x_tile'] == 1]

" Through all the coordinates for all tiles for an obsid into a Shapely object and let shapely define the boundary. This would be good to learn, b/c that’s exactly what shapely was made for and it is a standard GIS task to learn."

In [8]:
data[data['obsid'] == 'ESP_012079_0945'].x_tile.max()

5

In [9]:
allobs = data.obsid.unique() #give me every obsid

In [10]:

xmiba = []
xmaba = []
ymiba = []
ymaba = []

for obs in allobs:
    temp = data[data['obsid'] == obs] #look only at our obsid in question
    #Get xmins, xmaxes for the tiles
    xmi = temp.x_tile.min()
    xma = temp.x_tile.max()
    ymi = temp.y_tile.min()
    yma = temp.y_tile.max()
    xmiba.append(xmi)
    xmaba.append(xma)
    ymiba.append(ymi)
    ymaba.append(yma)
    
xmiba = np.asarray(xmiba)
ymiba = np.asarray(ymiba)
xmaba = np.asarray(xmaba)    
ymaba = np.asarray(ymaba)

Running and analyzing the above shows that x_tile has a min of 1, max of 5 and the y_tile has a min of 1, max of 145 across all our obsids.

In [11]:
results = 1

In [272]:
#similar to above, but now to create a dataframe which will help us get the corner values
bucket = []
for obs in allobs:
    d = {}
    temp = data[data['obsid'] == obs]
    d["x_tile_min"] = temp.x_tile.min()
    d["x_tile_max"] = temp.x_tile.max()
    d["y_tile_min"] = temp.y_tile.min()
    d["y_tile_max"] = temp.y_tile.max()
    d["obsid"] = obs
    d["top_left"] = (temp[(temp['x_tile'] == temp.x_tile.min()) & (temp["y_tile"] == temp.y_tile.min())].BodyFixedCoordinateX.values, temp[(temp['x_tile'] == temp.x_tile.min()) & (temp["y_tile"] == temp.y_tile.min())].BodyFixedCoordinateY.values)
    d["top_right"] = (temp[(temp['x_tile'] == temp.x_tile.min()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateX.values, temp[(temp['x_tile'] == temp.x_tile.min()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateY.values )
    d['bottom_left'] = (temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.min())].BodyFixedCoordinateX.values, temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.min())].BodyFixedCoordinateY.values)
    d['bottom_right'] = (temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateX.values, temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateY.values )
    bucket.append(d)
results = pd.DataFrame(bucket)


In [273]:
results

bottom_left  \
0      ([-67.69881540000002], [255.79257630000004])   
1             ([108.67046979999999], [457.5703882])   
2                                    ([nan], [nan])   
3           ([-319.50996000000004], [-323.8713523])   
4             ([227.52294890000002], [406.7287602])   
5                   ([-111.6923165], [131.7801493])   
6                                    ([nan], [nan])   
7                   ([-276.0157651], [-10.9464875])   
8              ([-157.25998380000001], [24.800988])   
9                    ([226.6364917], [406.2519646])   
10             ([-182.91961930000002], [1.4059434])   
11             ([110.992341], [461.28459630000003])   
12            ([-40.180741299999994], [165.442326])   
13                                   ([nan], [nan])   
14                   ([-156.1980398], [28.0622766])   
15            ([-278.076475], [-9.847341199999999])   
16   ([-269.95548019999995], [-13.376648499999998])   
17            ([470.2371477], [40.230123299999995])   
18    ([-272.78823539999996], [-8.263922299999999])   
19                                   ([nan], [nan])   
20           ([219.68874730000002], [-440.0846387])   
21                   ([-66.8270493], [258.3039062])   
22                   ([-155.9692195], [28.7835078])   
23           ([219.99603480000002], [-437.7755852])   
24                                   ([nan], [nan])   
25          ([-180.8096829], [-2.5581232999999997])   
26                   ([-79.4964947], [242.6512937])   
27            ([-280.90285439999997], [-8.4376082])   
28                   ([219.589494], [-451.1573348])   
29     ([-66.48888240000001], [259.28016809999997])   
..                                              ...   
191            ([193.55327140000003], [444.370916])   
192     ([7.642432799999999], [174.57109690000001])   
193           ([192.3731967], [443.81859000000003])   
194           ([112.26952050000001], [462.4600239])   
195                                  ([nan], [nan])   
196   ([221.21788709999998], [-439.97783839999994])   
197                                  ([nan], [nan])   
198   ([218.96548680000004], [-446.61933210000007])   
199         ([-52.519985999999996], [-283.5742963])   
200                      ([7.30397], [174.9633451])   
201           ([462.7555402], [42.420154700000005])   
202           ([264.230999], [-358.66229530000004])   
203          ([220.65749790000004], [-451.5382478])   
204           ([6.1815017999999995], [174.2590387])   
205                   ([38.4962776], [164.9726556])   
206           ([25.7848378], [-422.25858039999997])   
207           ([5.2699107000000005], [175.7059135])   
208          ([-114.0989393], [127.91868590000001])   
209          ([-3.3219762000000004], [168.2275885])   
210                    ([1.2026403], [166.1409396])   
211          ([-173.5054495], [26.849631899999995])   
212          ([-111.79683290000001], [131.8590622])   
213   ([220.55533250000002], [-447.84143150000006])   
214           ([112.15484409999999], [462.3659625])   
215           ([192.9338203], [443.35984160000004])   
216    ([187.83686819999997], [438.80692750000003])   
217           ([109.8677807], [458.81832639999993])   
218              ([193.63044], [444.5435811000001])   
219                  ([-10.9901075], [281.2380202])   
220          ([217.68302409999998], [-448.9035055])   

                                       bottom_right            obsid  \
0            ([-62.388268200000006], [269.3496037])  ESP_012079_0945   
1             ([121.8381508], [477.67216169999995])  ESP_011697_0980   
2                                    ([nan], [nan])  ESP_021455_0935   
3    ([-328.12714460000007], [-328.13045330000006])  ESP_012008_0975   
4             ([239.92119440000002], [417.6651198])  ESP_012436_0980   
5                   ([-103.0975151], [148.7454417])  ESP_020347_0930   
6                                    ([nan], [nan])  ESP_020519_0935   
7                    ([-288.3257547], [-3.5948949])  

Throw out NAN values, filter by the obsid's for each region, add that region name as a column so as to sort better.  
I think the bottom left and bottom right x values being diff. is okay, just curvature and that the tiles are mostly off-center from what I was thinking.

In [14]:
results.x_tile_min[0]

1

In [15]:
temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateX.values

array([221.6780802])

In [16]:
corners_df = results

In [17]:
corners_df.to_csv("Corners_df.csv")

Use olddata dataframe to find the obsids and their associated region names, add region names to appropriate obsids in newres and actually drop nan values

In [22]:
olddata.head()

Latitude       Ls  MY Which  fraction  n_objects            obsid  \
0  -85.4074  214.785  29  both     0.115        933  ESP_012079_0945   
1  -85.4065  183.727  29  both     0.070       1971  ESP_011407_0945   
2  -85.4056  198.040  29  both     0.072       1525  ESP_011723_0945   
3  -85.4049  207.892  29  both     0.168        788  ESP_011934_0945   
4  -85.4067  235.282  29  both     0.111       1285  ESP_012501_0945   

     pixel_area   region_name  
0  2.551911e+07  Macclesfield  
1  2.592302e+06  Macclesfield  
2  8.455380e+06  Macclesfield  
3  7.357966e+06  Macclesfield  
4  4.882627e+06  Macclesfield

In [23]:
olddata.set_index("region_name").head()

Latitude       Ls  MY Which  fraction  n_objects  \
region_name                                                      
Macclesfield  -85.4074  214.785  29  both     0.115        933   
Macclesfield  -85.4065  183.727  29  both     0.070       1971   
Macclesfield  -85.4056  198.040  29  both     0.072       1525   
Macclesfield  -85.4049  207.892  29  both     0.168        788   
Macclesfield  -85.4067  235.282  29  both     0.111       1285   

                        obsid    pixel_area  
region_name                                  
Macclesfield  ESP_012079_0945  2.551911e+07  
Macclesfield  ESP_011407_0945  2.592302e+06  
Macclesfield  ESP_011723_0945  8.455380e+06  
Macclesfield  ESP_011934_0945  7.357966e+06  
Macclesfield  ESP_012501_0945  4.882627e+06

In [24]:
allregions = olddata.region_name.unique() #give me every region name

In [61]:
bucket = []
for region in allregions:
    d = {}
    d['region_name'] = region
    d['obsids'] = olddata[olddata['region_name']==region].obsid.unique()
    bucket.append(d)
name_obs = pd.DataFrame(bucket)

In [26]:
name_obs

obsids            region_name
0   [ESP_012079_0945, ESP_011407_0945, ESP_011723_...           Macclesfield
1   [ESP_011697_0980, ESP_011341_0980, ESP_011842_...              Starburst
2   [ESP_012884_0935, ESP_012251_0935, ESP_012739_...      Manhattan_Classic
3                  [ESP_012008_0975, ESP_011296_0975]             Wellington
4   [ESP_012436_0980, ESP_012291_0980, ESP_012647_...                 Albany
5   [ESP_011565_0930, ESP_012211_0930, ESP_011776_...                 Bilbao
6   [ESP_011931_0945, ESP_012063_0945, ESP_011350_...                 Ithaca
7   [ESP_011960_0925, ESP_012817_0925, ESP_012316_...             Portsmouth
8   [ESP_012077_0930, ESP_012499_0930, ESP_011431_...                  Pisaq
9   [ESP_011711_0930, ESP_012990_0930, ESP_012634_...  Manhattan_Frontinella
10  [ESP_012227_0980, ESP_012860_0980, ESP_012939_...            BuenosAires
11  [ESP_011900_0985, ESP_012691_0985, ESP_012322_...       Inca_City_Ridges
12  [ESP_012256_0985, ESP_012744_0985, ESP_011491_...              Inca_City
13  [ESP_012344_0950, ESP_011447_0950, ESP_012704_...                   Giza
14  [ESP_011737_0980, ESP_012515_0980, ESP_011526_...                Potsdam
15  [ESP_012693_0950, ESP_012838_0950, ESP_013049_...                   Troy
16  [ESP_012463_0930, ESP_013030_0930, ESP_011606_...            Oswego_Edge
17  [ESP_012648_0930, ESP_012081_0930, ESP_011422_...                Halifax
18                                  [ESP_011680_1055]            Caterpillar
19  [ESP_012393_0965, ESP_012604_0965, ESP_012248_...              Rochester
20  [ESP_012821_0865, ESP_020532_0935, ESP_021468_...       Manhattan_Cracks
21                                  [ESP_011921_0950]            Schenectady
22                 [ESP_012254_1065, ESP_012887_1065]             Binghamton
23  [ESP_012038_0930, ESP_012671_0930, ESP_012816_...                   Atka
24                                  [ESP_011413_0970]               Cortland
25                                  [ESP_022193_0950]                Geneseo

Want to make it so every obsid in results gets checked against name_obs to find the region name, append the region name to the results dataframe in a new column titled region_name

In [53]:
name_obs[name_obs['region_name'] == 'Macclesfield'].obsids

0    [ESP_012079_0945, ESP_011407_0945, ESP_011723_...
Name: obsids, dtype: object

In [60]:
results[results['obsid']=='ESP_012079_0945']

bottom_left  \
0  ([-67.69881540000002], [-67.69881540000002])   

                             bottom_right            obsid  \
0  ([-62.388268200000006], [269.3496037])  ESP_012079_0945   

                         top_left  \
0  ([-68.3806735], [256.0561472])   

                                       top_right  x_tile_max  x_tile_min  \
0  ([-63.070566500000005], [269.61576610000003])           5           1   

   y_tile_max  y_tile_min  
0         109           1

In [72]:
results['region_name'] = np.nan #add nan values column so as to add real region names later.

In [274]:
#if its in name_obs.obsids, add a column to results with the region name of that one.
for region in allregions:
    x = name_obs[name_obs['region_name']== region].obsids  #collect every obsid in a region
    x = x.values.tolist()
    for vals in x[0]:
        results.loc[results.obsid == vals, 'region_name'] = region
#line above should replace region name with the proper region if the obsid matches any in x, but it doesnt work
results 

bottom_left  \
0      ([-67.69881540000002], [255.79257630000004])   
1             ([108.67046979999999], [457.5703882])   
2                                    ([nan], [nan])   
3           ([-319.50996000000004], [-323.8713523])   
4             ([227.52294890000002], [406.7287602])   
5                   ([-111.6923165], [131.7801493])   
6                                    ([nan], [nan])   
7                   ([-276.0157651], [-10.9464875])   
8              ([-157.25998380000001], [24.800988])   
9                    ([226.6364917], [406.2519646])   
10             ([-182.91961930000002], [1.4059434])   
11             ([110.992341], [461.28459630000003])   
12            ([-40.180741299999994], [165.442326])   
13                                   ([nan], [nan])   
14                   ([-156.1980398], [28.0622766])   
15            ([-278.076475], [-9.847341199999999])   
16   ([-269.95548019999995], [-13.376648499999998])   
17            ([470.2371477], [40.230123299999995])   
18    ([-272.78823539999996], [-8.263922299999999])   
19                                   ([nan], [nan])   
20           ([219.68874730000002], [-440.0846387])   
21                   ([-66.8270493], [258.3039062])   
22                   ([-155.9692195], [28.7835078])   
23           ([219.99603480000002], [-437.7755852])   
24                                   ([nan], [nan])   
25          ([-180.8096829], [-2.5581232999999997])   
26                   ([-79.4964947], [242.6512937])   
27            ([-280.90285439999997], [-8.4376082])   
28                   ([219.589494], [-451.1573348])   
29     ([-66.48888240000001], [259.28016809999997])   
..                                              ...   
191            ([193.55327140000003], [444.370916])   
192     ([7.642432799999999], [174.57109690000001])   
193           ([192.3731967], [443.81859000000003])   
194           ([112.26952050000001], [462.4600239])   
195                                  ([nan], [nan])   
196   ([221.21788709999998], [-439.97783839999994])   
197                                  ([nan], [nan])   
198   ([218.96548680000004], [-446.61933210000007])   
199         ([-52.519985999999996], [-283.5742963])   
200                      ([7.30397], [174.9633451])   
201           ([462.7555402], [42.420154700000005])   
202           ([264.230999], [-358.66229530000004])   
203          ([220.65749790000004], [-451.5382478])   
204           ([6.1815017999999995], [174.2590387])   
205                   ([38.4962776], [164.9726556])   
206           ([25.7848378], [-422.25858039999997])   
207           ([5.2699107000000005], [175.7059135])   
208          ([-114.0989393], [127.91868590000001])   
209          ([-3.3219762000000004], [168.2275885])   
210                    ([1.2026403], [166.1409396])   
211          ([-173.5054495], [26.849631899999995])   
212          ([-111.79683290000001], [131.8590622])   
213   ([220.55533250000002], [-447.84143150000006])   
214           ([112.15484409999999], [462.3659625])   
215           ([192.9338203], [443.35984160000004])   
216    ([187.83686819999997], [438.80692750000003])   
217           ([109.8677807], [458.81832639999993])   
218              ([193.63044], [444.5435811000001])   
219                  ([-10.9901075], [281.2380202])   
220          ([217.68302409999998], [-448.9035055])   

                                       bottom_right            obsid  \
0            ([-62.388268200000006], [269.3496037])  ESP_012079_0945   
1             ([121.8381508], [477.67216169999995])  ESP_011697_0980   
2                                    ([nan], [nan])  ESP_021455_0935   
3    ([-328.12714460000007], [-328.13045330000006])  ESP_012008_0975   
4             ([239.92119440000002], [417.6651198])  ESP_012436_0980   
5                   ([-103.0975151], [148.7454417])  ESP_020347_0930   
6                                    ([nan], [nan])  ESP_020519_0935   
7                    ([-288.3257547], [-3.5948949])  

In [282]:
bad = results.loc[6].bottom_left
print(bad)
print(type(bad))
bad2 = results.loc[6].bottom_right
print(bad2)
bad == bad2 #that's why
#why are the nan tuples not the same?! and how to fix?

(array([nan]), array([nan]))
<class 'tuple'>
(array([nan]), array([nan]))


False

In [265]:
indexNames = results[results['bottom_left'] == bad].index
indexNames

Int64Index([6], dtype='int64')

In [283]:
len(indexNames) #why doesnt this give all of them? because of bad !=bad2 above

1

In [258]:
results.drop(indexNames, inplace = True) #almost what I want
#need to figure out why indexNames doesnt find the other nan values
#once I do that the dataframe should be complete.

KeyError: '[2] not found in axis'

In [276]:
results

bottom_left  \
0      ([-67.69881540000002], [255.79257630000004])   
1             ([108.67046979999999], [457.5703882])   
2                                    ([nan], [nan])   
3           ([-319.50996000000004], [-323.8713523])   
4             ([227.52294890000002], [406.7287602])   
5                   ([-111.6923165], [131.7801493])   
6                                    ([nan], [nan])   
7                   ([-276.0157651], [-10.9464875])   
8              ([-157.25998380000001], [24.800988])   
9                    ([226.6364917], [406.2519646])   
10             ([-182.91961930000002], [1.4059434])   
11             ([110.992341], [461.28459630000003])   
12            ([-40.180741299999994], [165.442326])   
13                                   ([nan], [nan])   
14                   ([-156.1980398], [28.0622766])   
15            ([-278.076475], [-9.847341199999999])   
16   ([-269.95548019999995], [-13.376648499999998])   
17            ([470.2371477], [40.230123299999995])   
18    ([-272.78823539999996], [-8.263922299999999])   
19                                   ([nan], [nan])   
20           ([219.68874730000002], [-440.0846387])   
21                   ([-66.8270493], [258.3039062])   
22                   ([-155.9692195], [28.7835078])   
23           ([219.99603480000002], [-437.7755852])   
24                                   ([nan], [nan])   
25          ([-180.8096829], [-2.5581232999999997])   
26                   ([-79.4964947], [242.6512937])   
27            ([-280.90285439999997], [-8.4376082])   
28                   ([219.589494], [-451.1573348])   
29     ([-66.48888240000001], [259.28016809999997])   
..                                              ...   
191            ([193.55327140000003], [444.370916])   
192     ([7.642432799999999], [174.57109690000001])   
193           ([192.3731967], [443.81859000000003])   
194           ([112.26952050000001], [462.4600239])   
195                                  ([nan], [nan])   
196   ([221.21788709999998], [-439.97783839999994])   
197                                  ([nan], [nan])   
198   ([218.96548680000004], [-446.61933210000007])   
199         ([-52.519985999999996], [-283.5742963])   
200                      ([7.30397], [174.9633451])   
201           ([462.7555402], [42.420154700000005])   
202           ([264.230999], [-358.66229530000004])   
203          ([220.65749790000004], [-451.5382478])   
204           ([6.1815017999999995], [174.2590387])   
205                   ([38.4962776], [164.9726556])   
206           ([25.7848378], [-422.25858039999997])   
207           ([5.2699107000000005], [175.7059135])   
208          ([-114.0989393], [127.91868590000001])   
209          ([-3.3219762000000004], [168.2275885])   
210                    ([1.2026403], [166.1409396])   
211          ([-173.5054495], [26.849631899999995])   
212          ([-111.79683290000001], [131.8590622])   
213   ([220.55533250000002], [-447.84143150000006])   
214           ([112.15484409999999], [462.3659625])   
215           ([192.9338203], [443.35984160000004])   
216    ([187.83686819999997], [438.80692750000003])   
217           ([109.8677807], [458.81832639999993])   
218              ([193.63044], [444.5435811000001])   
219                  ([-10.9901075], [281.2380202])   
220          ([217.68302409999998], [-448.9035055])   

                                       bottom_right            obsid  \
0            ([-62.388268200000006], [269.3496037])  ESP_012079_0945   
1             ([121.8381508], [477.67216169999995])  ESP_011697_0980   
2                                    ([nan], [nan])  ESP_021455_0935   
3    ([-328.12714460000007], [-328.13045330000006])  ESP_012008_0975   
4             ([239.92119440000002], [417.6651198])  ESP_012436_0980   
5                   ([-103.0975151], [148.7454417])  ESP_020347_0930   
6                                    ([nan], [nan])  ESP_020519_0935   
7                    ([-288.3257547], [-3.5948949])  

the NAN NAN tuples are unique(Why??) so the above method won't work, come back to after consulting with michael

In [277]:
print(results.loc[0].bottom_left)
print(results.loc[0].bottom_left[0])
print(results.loc[0].bottom_left[1])


(array([-67.6988154]), array([255.7925763]))
[-67.6988154]
[255.7925763]
